app/cdap/components/RecipeList/ViewAllRecipes.tsx

```diff
@@ -14,32 +14,68 @@
  * the License.
  */
 
-import React from 'react';
+import React, { useState } from 'react';
 import T from 'i18n-react';
 import RecipeList from 'components/RecipeList';
-
 import Box from '@material-ui/core/Box';
 import { SortBy, SortOrder } from './types';
 import Breadcrumbs from '@material-ui/core/Breadcrumbs';
 import { Link } from 'react-router-dom';
 import Typography from '@material-ui/core/Typography';
 import { getCurrentNamespace } from 'services/NamespaceStore';
+import RecipeDetails from 'components/RecipeManagement/RecipeDetails';
+import { IRecipe, ActionType } from 'components/RecipeList/types';
+import DrawerWidget from 'components/common/DrawerWidget';
+import RecipeHeaderActionTemplate from 'components/RecipeManagement/RecipeDetails/HeaderActionTemplate';
 
 const PREFIX = 'features.WranglerNewUI.Recipe';
 
 const redirectToObj = `/ns/${getCurrentNamespace()}/wrangle`;
 
 const ViewAllRecipies = () => {
+  const [actionType, setActionType] = useState('');
+  const [isPanelOpen, setIsPanelOpen] = useState(false);
+  const [recipe, setRecipe] = useState<IRecipe>();
   const viewRecipeHandler = (selectedObject: any) => {
-    // To do : Integrate view recipe details panel
+    toggleOpen();
+    setRecipe(selectedObject);
+    setActionType(ActionType.VIEW_RECIPE);
   };
 
   const handleEditRecipe = (selectedObject: any) => {
     // To do : Integrate Edit recipe details panel
   };
 
+  const toggleOpen = () => setIsPanelOpen(!isPanelOpen);
+
+  const getChildComponent = () => {
+    if (actionType === ActionType.VIEW_RECIPE) {
+      return <RecipeDetails selectedRecipe={recipe} />;
+    }
+    if (recipe) {
+      return <></>; // TODO: Here we will render Edit Recipe component once we integrate edit recipe
+    }
+  };
+
   return (
     <>
+      <DrawerWidget
+        anchor="right"
+        closeClickHandler={toggleOpen}
+        headingText={
+          actionType === ActionType.VIEW_RECIPE
+            ? T.translate(`${PREFIX}.recipeDetails`)
+            : T.translate(`${PREFIX}.editRecipe`)
+        }
+        showBackIcon={false}
+        showDivider={Boolean(actionType === ActionType.VIEW_RECIPE)}
+        open={isPanelOpen}
+        headerActionTemplate={
+          actionType === ActionType.VIEW_RECIPE && <RecipeHeaderActionTemplate />
+        }
+        dataTestId={`${actionType}-drawer-widget`}
+        children={getChildComponent()}
```

Comment:
we should wrap the parent around the child component to pass child components

Reply:
removed children prop and wrapped inside parent